engine/internal/modeldownloader/huggingface/download.go

```diff
@@ -8,18 +8,40 @@ import (
 	"log"
 	"os"
 	"path/filepath"
+	"strings"
 
+	"github.com/aws/aws-sdk-go-v2/service/s3"
 	"github.com/aws/smithy-go"
 )
 
 const siFilename = "model.safetensors.index.json"
 
 type s3Client interface {
 	Download(ctx context.Context, f io.WriterAt, path string) error
+	ListObjectsPages(ctx context.Context, prefix string, f func(page *s3.ListObjectsV2Output, lastPage bool) bool) error
 }
 
 // DownloadModelFiles downloads model files from S3.
 func DownloadModelFiles(ctx context.Context, s3Client s3Client, srcS3Path string, destDir string) error {
+	keys, err := listFiles(ctx, s3Client, srcS3Path)
+	if err != nil {
+		return fmt.Errorf("list files: %s", err)
+	}
+	if len(keys) == 0 {
+		return fmt.Errorf("no objects found under %q", srcS3Path)
+	}
+
+	found := false
```

Comment:
Can you also add some comment on the rationale behind this change? If we only look for some specific objects, can get call GetObject?